gateway/api.go

```diff
@@ -1039,7 +1039,9 @@ func (gw *Gateway) handleGetAPIOAS(apiID string, modePublic bool) (interface{},
 
 	obj, code := gw.handleGetAPI(apiID, true)
 	if apiOAS, ok := obj.(*oas.OAS); ok && modePublic {
-		apiOAS.RemoveTykExtension()
+		oasCopy := *apiOAS
```

Comment:
Use [OAS.Clone](https://github.com/TykTechnologies/tyk/blob/master/apidef/oas/oas.go#L158) for a deep copy.